rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -559,6 +576,44 @@ impl FirmwareProcessor {
         (nb, nf)
     }
 
+    /// Read request from mailbox and verify the checksum
+    ///
+    /// # Arguments
+    /// * `txn` - Mailbox Receive Transaction
+    /// * `data` - Data buffer for the expected request
+    ///
+    /// # Returns
+    /// * `()` - Ok
+    ///    Error code on failure.
+    fn copy_req_verify_chksum(txn: &mut MailboxRecvTxn, data: &mut [u8]) -> CaliptraResult<()> {
+        // NOTE: Currently ROM only supports commands with a fixed request size
+        //       This check will need to be updated if any commands are added with a variable request size
+        if txn.dlen() as usize != data.len() {
+            return Err(CaliptraError::FW_PROC_MAILBOX_INVALID_REQUEST_LENGTH);
```

Comment:
Can we add a test case that checks for this error too?

Reply:
This is a good point, I can add that. Would it be alight to merge without it and I will add this week?

Reply:
Issue filed https://github.com/chipsalliance/caliptra-sw/issues/818